internal/buffer/cursor.go

```diff
@@ -193,7 +193,7 @@ func (c *Cursor) Deselect(start bool) {
 		if start {
 			c.Loc = c.CurSelection[0]
 		} else {
-			c.Loc = c.CurSelection[1].Move(-1, c.buf)
+			c.Loc = c.CurSelection[1]
```

Comment:
Now this also effectively brings back `MoveCursorDown(1)` which we agreed not to do (https://github.com/zyedidia/micro/pull/3540#discussion_r1855543349)...

Reply:
But implicit and not explicit...I didn't add any additional `MoveCursorDown(1)`. It's caused by the fact that `micro` cursor IS already placed at the next line, since the selection ends AFTER the newline.
We removed the evidence, which showed us this effect already, by removing the highlighting in the ruler.
Now we need to "workaround" this obvious effect again.